### ISSUE TITLE
[Feature request] Force update for tag

### ISSUE THREAD

**Issue by FTOcotral** [ForkIssues/TrackerWin#1582]
Hello,

First, thanks for your nice sofware.I noticed that I had to go back to console to force update a tag, as no tick like the branch present.
Can you add a force update button for tags please ?

**Comment by DanPristupov** [comment]
What will it do?
It would be great if you send a link to the git documentation.

**Comment by FTOcotral** [comment]
just to add the option -f on a push tag command. Just like it does for branch.

Use case :
Let's say you create a tag. Then push the tag to the server. Then delete your tag only locally without delete the distant one because whether you miss the button or you don't want to remove it for now. You create the tag on another commit. If you want know to push it again, it will be rejected if not forced.

**Comment by DanPristupov** [comment]
So, you ask for `force push`, not force update, right?

> just to add the option -f on a push tag command. Just like it does for branch.

Let's say you `force-pushed` a tag. What the other users are expected to do next? There is a local tag `tag1` on one commit, and remote `tag1` on another commit. Git fetch will not update that and show 'rejected: would clobber existing tag' error.

**Comment by FTOcotral** [comment]
> Git fetch will not update that and show 'rejected: would clobber existing tag' error.

You're right, I didn't think about it. So having a complete management would be a bit more complex, and user should be warn remote tag has change and have a prompt to request if he want to update the tag in which case a force fetch should be done.

But as you can work with other users who don't use Fork, you can't assume it will never happen, so it should be considered.

**Comment by dsbert** [comment]
Similar request here, but when fetching instead of pushing. Using `git fetch --tags -f`, the local tags are overwritten, and a subsequent `git fetch` will succeed.

**Comment by wienans** [comment]
@DanPristupov i am searching for a similar function.

i want to move the local tag to a newer commit. 

Current workflow: delete tag create a new tag with the same name.

But as this can cause typos and so on i would love to have the option to simply move the tag forward to a other commit.

With Gitkraken you can checkout the tag similar to checking out the branch and then use the same "Reset "BRANCH" to Here" to move the Tag. In case of current Version of Fork you only can do "Reset "HEAD" to Here"

Or reusing the new tag menu it warns you if the tag name already exists and block you from pressing "Create and Push". Maybe it is possible to place a Force checkbox to overwrite the Tag to the new if the box is checked similar to the force push with branches.

**Comment by nathan-coleman** [comment]
@DanPristupov I would really appreciate a force create checkbox right beside the existing push checkbox when creating a tag. Here is where I would want it:

![CreateTagImage](https://github.com/user-attachments/assets/821aef1c-d9a3-46f2-a962-47c266e864e3)

Additionally, as  @wienans mentioned, a "Reset 'TAG' to here" button as well as "Reset 'HEAD' to here" button might be valuable. I don't think it would be as important though. Mostly I just want a way to force create / push tags without falling back to the command line.
